GEOSwiftMapboxGL/MapboxGLAnnotations.swift

```diff
@@ -17,112 +24,93 @@ public protocol GEOSwiftMapboxGL {
-//                MKPolygon in
-//                return MKPolygonWithCoordinatesSequence(linearRing.points)
-//            })
-            
-            let polygon = MGLPolygon(coordinates: &exteriorRingCoordinates, count: UInt(exteriorRingCoordinates.count) /*, interiorPolygons: interiorRings*/)
-            return polygon
-                                     
-        case is MultiPolygon<Polygon>:
-          let mglPolygons = (self as! MultiPolygon).geometries.map({ (polygon: Polygon) -> MGLPolygon in
-            return polygon.mapboxShape() as! MGLPolygon
-          })
-          return MGLMultiPolygon(polygons: mglPolygons) 
-                                     
-        default:
-            let geometryCollectionOverlay = MGLShapesCollection(geometryCollection: (self as! GeometryCollection))
-            return geometryCollectionOverlay
-        }
-    }
+extension MGLPointAnnotation {
+  convenience init(_ coord: CLLocationCoordinate2D) {
+    self.init()
+    self.coordinate = coord
+  }
+  
+  convenience init(point: Point) {
+    self.init(CLLocationCoordinate2D(point))
+  }
 }
 
-private func MGLPolygonWithCoordinatesSequence(coordinates: CoordinatesCollection) -> MGLPolygon {
-    var coordinates = coordinates.map({ (point: Coordinate) ->
-        CLLocationCoordinate2D in
-        return CLLocationCoordinate2DFromCoordinate(point)
-    })
-    return MGLPolygon(coordinates: &coordinates,
-        count: UInt(coordinates.count))
-    
+extension MGLPolyline {
+  convenience init(lineString: LineString) {
+    var points = lineString.points.map(CLLocationCoordinate2D.init)
+    self.init(coordinates: &points, count: UInt(points.count))
+  }
 }
 
-/**
-MGLShape subclass for GeometryCollections.
-The property `shapes` contains MGLShape subclasses instances. When drawing shapes on a map be careful to the fact that that these shapes could be overlays OR annotations.
-*/
-public class MGLShapesCollection : MGLShape, MGLOverlay {
-    let shapes: Array<MGLShape>
-    public let centroid: CLLocationCoordinate2D
-    public let overlayBounds: MGLCoordinateBounds
-    
-    // inserting the where clause in the following generic create some confusion in the precompiler that raise the following error:
-    // Cannot invoke initializer for type ... with an argument list of type (geometryCollection: GeometryCollection<T>)
-    // 1. Expected an argument list of type (geometryCollection: GeometryCollection<T>)
-    required public init<GEOSwiftMapboxGL>(geometryCollection: GeometryCollection<GEOSwiftMapboxGL>) {
-        let shapes = geometryCollection.geometries.map({ (geometry: GEOSwiftMapboxGL) ->
-            MGLShape in
-            return geometry.mapboxShape()
-        })
+extension MGLPolygon {
+  convenience init(linearRing: Polygon.LinearRing) {
+    var coordinates = linearRing.points.map(CLLocationCoordinate2D.init)
+    self.init(coordinates: &coordinates, count: UInt(coordinates.count))
+  }
+  
```

Comment:
I'd remove this. Anyone who needs this can do it via `init(polygon:) with composition.